### PR TITLE
Submission hotfixes

### PR DESCRIPTION
**Pull request by JamesGlover**

### CODE REVIEW

File: app/api/model_extensions/batch.rb [hunk 1]
```diff
@@ -5,9 +5,6 @@
 module ModelExtensions::Batch
   def self.included(base)
     base.class_eval do
-      # These were in Batch but it makes more sense to keep them here for the moment
```
Comment (JamesGlover) [inline app/api/model_extensions/batch.rb]: Not really, no. I've moved them back onto batch.

File: app/models/aliquot.rb [hunk 2]
```diff
@@ -9,6 +9,8 @@ class Aliquot < ActiveRecord::Base
   include Api::Messages::FlowcellIO::AliquotExtensions
   include AliquotIndexer::AliquotScopes
 
+  TAG_COUNT_NAMES = ['Untagged','Single','Dual']
+
```
Comment (JamesGlover) [inline app/models/aliquot.rb]: Tag number was confusing, as it was unclear if it applied to per sample, or per pool. I've opted for names instead, to make it more apparent.

File: app/controllers/bulk_submissions_controller.rb [hunk 3]
```diff
@@ -12,10 +12,13 @@ def index
```
Comment (JamesGlover) [inline app/controllers/bulk_submissions_controller.rb]: Submission Templates have been added to the bulk submissions page as a list. I'll tweak this to be a before filter.

File: app/models/batch.rb [hunk 4]
```diff
@@ -7,10 +7,28 @@
 require 'aasm'
 
 class Batch < ActiveRecord::Base
```
Comment (JamesGlover) [inline app/models/batch.rb]: Using proper rails relationships here.

File: app/models/role.rb [hunk 5]
```diff
@@ -37,32 +37,21 @@ def self.included(base)
       base.extend(ClassMethods)
```
Comment (JamesGlover) [inline app/models/role.rb]: Modernised the scopes. Rails wasn't able to order the old version properly as it couldn't generate the correct SQL

File: app/views/bulk_submissions/new.html.erb [hunk 6]
```diff
@@ -12,7 +12,7 @@
       <%= form.input :spreadsheet, :as => :file, :label => 'Spreadsheet' %>
```
Comment (JamesGlover) [inline app/views/bulk_submissions/new.html.erb]: Add list of templates, and prettyfy the page

File: app/views/pipelines/_requests.html.erb [hunk 7]
```diff
@@ -19,14 +19,13 @@
         <th><%= link_to "Request ID", "javascript:void(0);" %></th>
         <th><%= link_to "Name", "javascript:void(0);" %></th>
         <th><%= link_to "Request comments", "javascript:void(0);" %></th>
-        <th><%= link_to "Asset type", "javascript:void(0);" %></th>
```
Comment (JamesGlover) [inline app/views/pipelines/_requests.html.erb]: Asset type is now redundant and misleading. I've removed it.

File: app/models/role.rb [hunk 8]
```diff
@@ -37,32 +37,21 @@ def self.included(base)
...
-       scope :with_related_users_included, -> { includes(:roles => :users ) }
-       scope :of_interest_to, ->(user) {
-          {
-            :joins => joins_through_to_users,
-            :conditions => ['rj_u.id=?', user.id],
-            :group => "rj_r.authorizable_id"
-          }
+        scope :with_related_users_included, -> { includes(:roles => :users ) }
+        scope :of_interest_to, ->(user) {
+          joins(:users).where(users:{id:user}).uniq
```
Comment (emrojo) [inline app/models/role.rb]: I think it should be 'user.id' like: joins(:users).where(users:{id:user.id}).uniq

Comment (emrojo) [inline app/models/role.rb]: Ok, it is converted by rails automatically.

**Pull request merged**